runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ExternalTest.java

```diff
@@ -139,7 +139,7 @@ public void expandPythonTest() {
           testPipeline
               .apply(Create.of("1", "2", "2", "3", "3", "3"))
               .apply(
-                  External.<KV<String, Integer>>of(
+                  External.<PCollection<String>, KV<String, Integer>>of(
```

Comment:
From a usability point of view, is it possible to infer this first type argument (as it must match the input type)? Presumably we were inferring String before...

Reply:
It's generally impossible to only infer the first type parameter when the second type parameter needs to be explicitly defined. There's a workaround however. I updated the example.

Reply:
Hmm... That's unfortunate.

I wonder if we should have separate methods for the common "applies to a single collection" and more generic case.

Reply:
Yeah, we can create a variant of `of()` for the common case. Do you have any good naming suggestion? :)